### PR TITLE
CT 107: Add "School Locations" section to the Institution Profile Page #19447

### PR DESCRIPTION
**Pull request by zurbergram**
## Description:
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/19447

As a comparison tool user, I need to be able to review all locations that are in an institution network so that I can easily compare the housing benefits by location within the network.

## Testing Done:
- local testing
- unit testing

## Acceptance Criteria:
- [ ] There is a "School locations" section between the "Student outcomes" and "Cautionary information" sections.
- [ ] The beginning of the section displays the following informational content: "Below are locations for [Main Campus Name].  Select a link to view another location and calculate the benefits you’d receive there."
    - The [Main Campus Name] is the name of the main campus (for example, if viewing a branch campus profile, it will display the parent main campus)
- [ ] After the opening information content a table is displayed with the following columns:
    - School name
    - Location
    - Estimated housing
- [ ] The "School name" column is populated with the "institution name" from the weams.csv.
- [ ] If the weams.csv row does not have a parent facility code, it is considered the main campus and "(Main campus)" is appended to the end of the "School name".
- [ ] The "Location" column is populated with the "City, State ZIP" using the following columns from the weams.csv:
    - physical institution city
    - physical institution state
    - physical institution zip code
- [ ] The "Estimated housing" column is populated with the housing rate based on the answer to the "Did you use your Post-9/11 GI Bill benefits for tuition, housing, or books for a term that started before January 1, 2018?" question:
     - If the answer is YES, the VA rate is displayed
     - If the answer is NO, the DOD rate is displayed.
- [ ] The first data row displays the main campus information.
- [ ] Rows following the main campus row are displayed alphabetically.
- [ ] Rows following display branch campuses (as indicated by a B in the weams.csv campus_type column) with their corresponding extensions (as indicated by a E in the weams.csv campus_type column) nested underneath.
- [ ] If there are more than 10 campuses displayed, the remainder are hidden.
- [ ] The user can click a "View more..." link to display the remaining campuses on the page.
- [ ] If a main campus has no associated branches or extensions, the "School locations" section is not displayed.
- [ ] The main/branch campus names are links to their corresponding profile pages if they are not the current profile page.
- [ ] If the campus profile page is currently being viewed, it is displayed as bold text.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by sonntag-adam — APPROVED** [review]

File: src/applications/gi/components/profile/SchoolLocations.jsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+    return (
+      <a href={`${facilityCode}${query}`}>
+        <h6>{name}</h6>
+      </a>
+    );
+  };
+
+  handleViewMoreClicked = () => {
+    this.setState({ ...this.state, viewMore: true });
```
Comment (rianfowler) [inline src/applications/gi/components/profile/SchoolLocations.jsx]: `setState` merges the new values with the existing state so `...this.state` shouldn't be necessary.

**Review by rianfowler — APPROVED** [review]

**Comment by zurbergram** [comment]
waiting on https://github.com/department-of-veterans-affairs/gibct-data-service/pull/411 to be merged first before merging this

File: src/applications/gi/components/profile/SchoolLocations.jsx (hunk 1 above)
Comment (zurbergram) [reply src/applications/gi/components/profile/SchoolLocations.jsx]: changed

**Pull request merged**